apiml-common/src/main/java/org/zowe/apiml/product/eureka/client/ApimlPeerEurekaNode.java

```diff
@@ -560,13 +560,15 @@ private static boolean isSuccess(int statusCode) {
          * @return true, if it is a network timeout, false otherwise.
          */
         private static boolean isNetworkConnectException(Throwable e) {
-            do {
-                if (e instanceof IOException && !(e instanceof SSLException)) {
-                    return true;
-                }
-                e = e.getCause();
-            } while (e != null);
-            return false;
+            if (e instanceof IOException && !(e instanceof SSLException)) {
+                return true;
+            }
+
+            var cause = e.getCause();
+            if ((cause == null) || (cause == e)) {
+                return false;
+            }
+            return isNetworkConnectException(cause);
```

Comment:
do we really need to do it recursive?

Reply:
ok typically teams decide whether to prefer non-recursiveness (it's usually not the most readable option) but since we don't have such an agreement it's ok I suppose